pkg/controller/core/clusterqueue_controller.go

```diff
@@ -497,3 +500,18 @@ func (r *ClusterQueueReconciler) updateCqStatusIfChanged(
 	}
 	return nil
 }
+
+func (r *ClusterQueueReconciler) getWorkloadsStatus(cq *kueue.ClusterQueue) *kueue.ClusterQueuePendingWorkloadsStatus {
+	pendingWorkloadsStatus := cq.Status.DeepCopy().PendingWorkloadsStatus
```

Comment:
I don't see a need for DeepCopy. For comparison we can use status, for setting a new structure we can create a new one and assign.